cmd/thanos/sidecar.go

```diff
@@ -458,3 +400,29 @@ func (s *promMetadata) Timestamps() (mint int64, maxt int64) {
 
 	return s.mint, s.maxt
 }
+
+type sidecarConfig struct {
+	http         httpConfig
+	grpc         grpcConfig
+	prometheus   prometheusConfig
+	connection   connectionConfig
+	tsdb         tsdbConfig
+	reloader     reloaderConfig
+	objStore     extflag.PathOrContent
+	shipper      shipperConfig
+	limitMinTime thanosmodel.TimeOrDurationValue
+}
+
+func (sc *sidecarConfig) registerFlag(cmd *kingpin.CmdClause) *sidecarConfig {
+	sc.http.registerFlag(cmd)
+	sc.grpc.registerFlag(cmd)
+	sc.prometheus.registerFlag(cmd)
+	sc.connection.registerFlag(cmd)
+	sc.tsdb.registerFlag(cmd)
+	sc.reloader.registerFlag(cmd)
+	sc.objStore = *regCommonObjStoreFlags(cmd, "", false)
+	sc.shipper.registerFlag(cmd)
+	sc.limitMinTime = *thanosmodel.TimeOrDuration(cmd.Flag("min-time", "Start of time range limit to serve. Thanos sidecar will serve only metrics, which happened later than this value. Option can be a constant time in RFC3339 format or time duration relative to current time, such as -1d or 2h45m. Valid duration units are ms, s, m, h, d, w, y.").
```

Comment:
@bwplotka are you happy with how this and the one above it are being parsed at this stage?

Reply:
actually I see `limitMinTime` isn't being set correctly yet